packages/core/lib/Interfaces/DSN.ts

```diff
@@ -0,0 +1,47 @@
+import { SentryError } from '../Sentry';
+
+interface IDSNParts {
+  source: string;
+  protocol: string;
+  user: string;
+  pass: string;
+  host: string;
+  port: string;
+  path: string;
+}
+
+export class DSN {
+  private dsnString: string;
+  private dsn: IDSNParts;
+  private dsnRegex = /^(?:(\w+):)\/\/(?:(\w+)(:\w+)?@)([\w\.-]+)(?::(\d+))?(\/.*)/;
```

Comment:
This could also be a (possibly exported) constant in this file. It seems weird to have it as an instance field.

Reply:
I would make it "public" when we actually need it somewhere else.